src/h5web/providers/hsds/api.ts

```diff
@@ -112,13 +125,27 @@ export class HsdsApi implements ProviderAPI {
 
   /* Processing methods to fetch links and attributes in addition to the entity. Also updates API members. */
 
+  private async processAttributes(
+    collection: HDF5Collection,
+    id: HDF5Id
+  ): Promise<HDF5Attribute[]> {
+    const { attributes: bareAttributes } = await this.fetchAttributes(
+      collection,
+      id
+    );
+    const valuePromises = bareAttributes.map(attr =>
+      this.fetchAttributeWithValue(collection, id, attr.name)
+    );
+    return Promise.all(valuePromises);
+  }
+
```

Comment:
I put the fetch in the `processAttributes` method to be consistent with other `process...` methods.

Reply:
Yeah, I think it makes sense!

Reply:
Finally, the complete fetch (attributes + values) was put in `fetchAttributes`